sdks/java/extensions/sql/build.gradle

```diff
@@ -53,10 +53,12 @@ def hadoopVersions = [
 hadoopVersions.each {kv -> configurations.create("hadoopVersion$kv.key")}
 
 dependencies {
+  // Required for processing Calcite's immutable annotations
```

Comment:
Odd, it seems like this is a dependency of the vendored_calcite_1_28_0 package and should be declared there? (The linkage check passed, so this is fine with me.)

Reply:
Yeah, to be honest I don't quite understand why this is necessary. Without this we get this error during annotation processing:
```
error: cannot access Value
  class file for org.immutables.value.Value not found
```

and then a series of errors due to missing AutoValue-generated classes like:
```
sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/TableName.java:73: error: cannot find symbol
    return new AutoValue_TableName(path == null ? Collections.emptyList() : path, tableName);
               ^
  symbol:   class AutoValue_TableName
  location: class TableName
```

Perhaps this is a bad interaction with another annotation processor?

I'm also inclined to just not worry about this, but I'm a little nervous we'll inflict the same annotation processing issue on our users.

Reply:
Ok to be sure I ran internal tests on this, which including building and running some templates. The templates didn't need to add `org.immutables.value:value` in order to build, so it looks like we won't be passing this issue along.

Reply:
If its only necessary for annotation processing consider adding it under `annotationProcessor` and possibly `compileOnly` (only if its a dep as well) like [BeamModulePlugin.groovy](https://github.com/apache/beam/blob/48318dd3aedeb208648149eca36165bbfff09e1a/buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy#L966) does.

Reply:
Hm well adding it under `annotationProcessor` doesn't alleviate the issue, but `compileOnly` (by itself) does.